upstream.go

```diff
@@ -59,6 +59,10 @@ type upstreamConn struct {
 
 	saslClient  sasl.Client
 	saslStarted bool
+
+	// set of pending LIST requests, per downstream
```

Comment:
```go
// set of LIST commands in progress, per downstream
```

to make it clear what "pending" means (sent but not finished, as opposed to not yet sent)